models/_sector/dex/trades/optimism/platforms/chainhop_optimism_base_trades.sql

```diff
@@ -0,0 +1,60 @@
+{{ config(
+    schema = 'chainhop_optimism'
+    ,alias = 'base_trades'
+    ,materialized = 'incremental'
+    ,file_format = 'delta'
+    ,incremental_strategy = 'merge'
+    ,unique_key = ['tx_hash', 'evt_index']
+    ,incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.block_time')]
+    )
+}}
+
+{% set chainhop_optimism_evt_trade_tables = [
+    source('chainhop_optimism', 'TransferSwapper_evt_DirectSwap')
+] %}
+
+with dexs AS (
+    {% for evt_trade_table in chainhop_optimism_evt_trade_tables %}
+        SELECT
+            evt_block_time          AS block_time,
+            evt_tx_to               AS taker,
+            evt_tx_from             AS maker,
+            amountIn                AS token_bought_amount_raw,
+            amountOut               AS token_sold_amount_raw,
+            CAST(NULL AS double)    AS amount_usd,
+            tokenIn                 AS token_bought_address,
+            tokenOut                AS token_sold_address,
+            contract_address        AS project_contract_address,
+            evt_tx_hash             AS tx_hash,
+            evt_block_number        AS block_number,
+            evt_index
+        FROM {{ evt_trade_table }}
+        {% if is_incremental() %}
+        WHERE {{incremental_predicate('evt_block_time')}}
+        {% endif %}
+
+        {% if not loop.last %}
+        UNION ALL
+        {% endif %}
+
+    {% endfor %}
+)
+
+SELECT
+    'optimism' AS blockchain,
+    'chainhop' AS project,
```

Comment:
Hi @ARDev097,
if `chainhop` and `saddle_finance` are newly added dexes, we may need to add their information in `dex.info` table:

https://github.com/duneanalytics/spellbook/blob/main/models/dex/dex_info.sql